### PR TITLE
Remove all code and references for ChurnBuster

### LINKED ISSUE #209: Remove ChurnBuster integration

### PR DESCRIPTION
**Pull request by garciajordy**
fix #209 

# Remove ChurnBuster Integration

### **Removed Files:**
- app/decorators/models/solidus_subscriptions/spree/wallet_payment_source/report_default_change_to_subscriptions.rb
- app/subscribers/solidus_subscriptions/churn_buster_subscriber.rb
- spec/lib/solidus_subscriptions_spec.rb
- spec/lib/solidus_subscriptions/churn_buster/client_spec.rb
- spec/models/spree/wallet_payment_source_spec.rb
- spec/subscribers/solidus_subscriptions/churn_buster_subscriber_spec.rb
- spec/support/vcr.rb

**solidus_subscriptions/lib/solidus_subscriptions/churn_buster Folder**
- client.rb
- order_serializer.rb
- serializer.rb
- subscription_customer_serializer.rb
- subscription_payment_method_serializer.rb
- subscription_serializer.rb

**solidus_subscriptions/spec/fixtures/cassettes Folder**
- churn_buster.yml

### **Changed Files:**
**README.md:**
- Removed ChurnBuster integration section.

**solidus_subscriptions.gemspec:**
- Remove "vcr" from development dependecy.
- Remove "webmock" from development dependency.

**app/models/solidus_subscriptions/subscription.rb:**
- Remove ".with_default_payment_source" scope.
- Remove
```
384-      if previous_changes.key?('payment_source_id') || previous_changes.key?('payment_source_type') || previous_changes.key?('payment_method_id')
385-        ::Spree::Event.fire(
386-          'solidus_subscriptions.subscription_payment_method_changed',
387-          subscription: self,
388-        )
389-      end
```

**config/initializers/subscribers.rb:**
- Remove churn_buster_subscriber.

**lib/solidus_subscriptions.rb:**
- Remove all the require paths for churn_buster.
- Remove the 'churn_buster' method.

**lib/solidus_subscriptions/configuration.rb:**
- Remove "churn_buster_account_id" and "churn_buster_api_key" from attr_accessor.
- Remove the "churn_buster?" method.

**dispatcher/payment_failed_dispatcher.rb:**
- Remove the Spree::Event.fire.

**dispatcher/success_dispatcher.rb:**
- Remove the Spree::Event.fire.

**spec/lib/solidus_subscriptions/dispatcher/payment_failed_dispatcher_spec.rb:**
- Remove the test for installments_failed_payment event.

**spec/lib/solidus_subscriptions/dispatcher/success_dispatcher_spec.rb:**
- Remove the test for installments_succeeded event.

**spec/models/solidus_subscriptions/subscription_spec.rb:**
- Remove test for tracking payment method changes.

### CODE REVIEW

**Comment by garciajordy** [comment]
@aldesantis

File: app/models/solidus_subscriptions/subscription.rb [hunk 1]
```diff
@@ -380,13 +376,6 @@ def emit_events_for_update
           subscription: self,
         )
       end
-
-      if previous_changes.key?('payment_source_id') || previous_changes.key?('payment_source_type') || previous_changes.key?('payment_method_id')
-        ::Spree::Event.fire(
-          'solidus_subscriptions.subscription_payment_method_changed',
-          subscription: self,
-        )
-      end
```
Comment (aldesantis) [inline app/models/solidus_subscriptions/subscription.rb:389]: We should keep all the `Spree::Event.fire` calls in place. Users of the extension can listen to those events to fire their own logic — they are not specific to the ChurnBuster integration, even though the integration did make use of them.

**Review by aldesantis — CHANGES_REQUESTED** [review]

File: app/models/solidus_subscriptions/subscription.rb (hunk 1 above)
Comment (garciajordy) [reply app/models/solidus_subscriptions/subscription.rb]: Done!

**Comment by garciajordy** [comment]
# Changes

**app/models/solidus_subscriptions/subscription.rb:**
- Add removed ```Spree::Event.fire``` as users of the extension can listen to those events to fire their own logic.

_Thanks for taking the time to review my changes_ 😄

**Comment by kennyadsl** [comment]
Closing for now, we can reopen if we want to move this forward.

**Pull request closed without merging**